### PR TITLE
cleanup of conda related scripts that are no longer needed

### PR DESCRIPTION
**Pull request by shaikh-rashid**
This PR replaces the use of `conda` with `pipenv`. Users will no longer have to install binaries or maintain large conda environment. As part of the change, base install will no longer require the use `make bin-all` or `make init-conda`, both have been replaced by `make init-pipenv`.

Testing should be done by running the following commands:
Remove miniconda and related miniconda config from `~/.bashrc`
`make bin-pyenv`
`which pyenv || exec bash`
`make init-pipenv`
`pipenv shell`
`make init-docker`
`make compose`
`make init-authx`

Reviewers should test on the folllowing platforms:
- [x] Windows
- [x] Linux
- [x] Mac OS X
- [ ] Mac OS X (M1)

### CODE REVIEW

**Review by daisieh — CHANGES_REQUESTED** [review]
If my system doesn't have the current VENV_PYTHON (right now, 3.10), it fails.

```
(candig) daisieh2019:CanDIGv2$ make init-pipenv
echo "    started init-pipenv" >> /Users/daisie/Documents/Work/Sandbox/candig/CanDIGv2/tmp/progress.txt
`pip3 install --user pipenv`
bash: Collecting: command not found
make: [init-pipenv] Error 127 (ignored)
pipenv --python 3.10
Loading .env environment variables...
Warning: Python 3.10 was not found on your system...
Neither 'pyenv' nor 'asdf' could be found to install Python.
You can specify specific versions of Python with:
$ pipenv --python path/to/python
make: *** [init-pipenv] Error 1```

**Comment by shaikh-rashid** [comment]
@daisieh looks like you have conda still active while using pipenv. I think you will need to deactivate it and try again.

**Comment by daisieh** [comment]
Deactivate or uninstall? Because deactivate did not solve the problem.

**Comment by daisieh** [comment]
```
(candig) daisieh2019:CanDIGv2$ conda deactivate
(base) daisieh2019:CanDIGv2$ make init-pipenv
echo "    started init-pipenv" >> /Users/daisie/Documents/Work/Sandbox/candig/CanDIGv2/tmp/progress.txt
`pip3 install --user pipenv`
bash: Collecting: command not found
make: [init-pipenv] Error 127 (ignored)
pipenv --python 3.10
Loading .env environment variables...
Warning: Python 3.10 was not found on your system...
Neither 'pyenv' nor 'asdf' could be found to install Python.
You can specify specific versions of Python with:
$ pipenv --python path/to/python
make: *** [init-pipenv] Error 1
```

**Comment by shaikh-rashid** [comment]
@daisieh you still have the base conda env activated.

**Comment by daisieh** [comment]
I had to comment out a bunch of stuff in my .bash_profile to make conda go away, but I am still getting errors:
```
daisieh2019:CanDIGv2$ make init-pipenv
echo "    started init-pipenv" >> /Users/daisie/Documents/Work/Sandbox/candig/CanDIGv2/tmp/progress.txt
`pip3 install --user pipenv`
  WARNING: The script virtualenv-clone is installed in '/Users/daisie/Library/Python/3.9/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script virtualenv is installed in '/Users/daisie/Library/Python/3.9/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts pipenv and pipenv-resolver are installed in '/Users/daisie/Library/Python/3.9/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
WARNING: There was an error checking the latest version of pip.
bash: Collecting: command not found
make: [init-pipenv] Error 127 (ignored)
pipenv --python 3.10
Loading .env environment variables...
Warning: Python 3.10 was not found on your system...
Neither 'pyenv' nor 'asdf' could be found to install Python.
You can specify specific versions of Python with:
$ pipenv --python path/to/python
make: *** [init-pipenv] Error 1
daisieh2019:CanDIGv2$ 
```

**Comment by shaikh-rashid** [comment]
@daisieh the warning indicates that you do not have your python virtualenv and host python set up. You will need to resolve the issues regarding PATH and try again.

**Comment by daisieh** [comment]
But isn't the idea of this target supposed to be to make the setup automatic?

File: Makefile [hunk 1]
```diff
@@ -34,41 +31,6 @@ mkdir:
...
-.PHONY: bin-all
-bin-all: bin-conda
-
-
-#>>>
-# download miniconda package
-# make bin-conda
-
-#<<<
-bin-conda: mkdir
```
Comment (kcranston) [inline Makefile:51]: this target needs to be replicated in init-pipenv, else you don't have a `tmp` dir for the logfile (among other paths):
`bash: /Users/karen/Documents/candig/repos/CanDIGv2/tmp/progress.txt: No such file or directory`

**Comment by kcranston** [comment]
After resolving the lack of `tmp` dir by running `make mkdir` manually, I get these errors:

```
(base) $  CanDIGv2 git:(conda-venv) make init-pipenv
echo "    started init-pipenv" >> /Users/karen/Documents/candig/repos/CanDIGv2/tmp/progress.txt
`pip3 install --user pipenv`
  WARNING: The script virtualenv-clone is installed in '/Users/karen/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script virtualenv is installed in '/Users/karen/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts pipenv and pipenv-resolver are installed in '/Users/karen/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
bash: Collecting: command not found
make: [init-pipenv] Error 127 (ignored)
pipenv --python 3.10
bash: pipenv: command not found
make: *** [init-pipenv] Error 127
```

**Comment by kcranston** [comment]
What does `pipenv` give us that the default `venv` does not?

If we are going to use pipenv instead of venv (and I am not yet convinced), then we should also use a Pipfile isntead of requirements.txt.

File: Makefile (hunk 1 above)
Comment (shaikh-rashid) [reply Makefile:51]: Fixed with latest push

**Comment by daisieh** [comment]
Is there a reason to *replace* conda with either pipenv or venv, or can we just have all of them as options?

**Review by daisieh — APPROVED** [review]
Except for the weird edge case that I ran into that seems to have had nothing to do with this branch or changes (and resolved itself with a few clean/inits), this works fine and much faster than conda. Ran through init-authx successfully and was able to clean successfully as well.

**Comment by kcranston** [comment]
The `bin-pyenv` step writes some configuration to `.bashrc`. If you are using a different shell (e.g. zsh is default on newer macs), later steps fail because this needs to be replicated in `.zshrc`. So, this PR should include documentation on a workaround for zsh (either instructions for adding the config to `.zshrc` and / or adding the filename of the login script as a variable).

**Review by OrdiNeu — APPROVED** [review]
Works for me, tested a while ago

**Review by CourtneyGosselin — APPROVED** [review]
LGTM

**Pull request merged**